### PR TITLE
vweb: add docs for [vweb_global] and shared attributes.

### PR DESCRIPTION
**Pull request by Casper64**
Add documentation about the use of `[vweb_global]`, shared fields and their pros/cons

### <samp>🤖 Generated by Copilot at 3508f24</samp>

This pull request updates the `vlib/vweb/README.md` file to include a new section on how to use the `[vweb_global]` attribute and `shared` fields for database pooling in `vweb` apps. This change aims to improve the documentation and provide examples for common scenarios.

### <samp>🤖 Generated by Copilot at 3508f24</samp>

* Add a new section to the `vweb` README.md file to explain how to use the `[vweb_global]` attribute and `shared` fields for data persistence and modification across requests ([link](https://github.com/vlang/v/pull/18098/files?diff=unified&w=0#diff-0363f38173fe973cfa73ea26ebe6ed582a470e86ad0ef0b249acde5755c22f80R485-R589))

### CODE REVIEW

File: vlib/vweb/README.md [hunk 1]
```diff
@@ -482,6 +482,111 @@ fn main() {
...
+	return app.text('Total visitors: ${count}')
+}
+```
+
+#### Drawback of Shared Objects
+The drawback of using shared objects is that it affects performance. In the previous example
+`App.counter` needs to be locked each time the page is loaded if there are simultaneous
+requests the next requests will have to wait for the lock to be released.
+In the context of the previous example, it would be better to make a call to a database 
+to update and get the visitor count.
```
Comment (spytheman) [inline vlib/vweb/README.md:586]: That is misleading - the overhead of calling to an even an embedded DB like sqlite, is much greater than a lock.

File: vlib/vweb/README.md [hunk 2]
```diff
@@ -482,6 +482,111 @@ fn main() {
 If you don't use the default number of workers (`nr_workers`) you have to change 
 it to the same number in `vweb.run_at` as in `vweb.database_pool`
 
+### Extending the App struct with `[vweb_global]`
+You can extend the `App` struct however you like, but there are some things you
+have to keep in mind. Under the hood at each request a new instance of `App` is
```
Comment (spytheman) [inline vlib/vweb/README.md:487]:
```suggestion
You can change your `App` struct however you like, but there are some things you
have to keep in mind. Under the hood at each request a new instance of `App` is
```

File: vlib/vweb/README.md (hunk 1 above)
Comment (Casper64) [reply vlib/vweb/README.md:586]: Is the sentence below it ok to be kept in? "It is best practice to limit the use of shared objects as much as possible."

Comment (spytheman) [reply vlib/vweb/README.md:586]: Yes, dealing with shared state is pain in general, even with the enforced checking that V does.

**Review by spytheman — APPROVED** [review]

**Pull request merged**